kcron-core/src/commonMain/kotlin/com/ucasoft/kcron/core/builders/Builder.kt

```diff
@@ -27,35 +27,35 @@ class Builder(firstDayOfWeek: WeekDays = WeekDays.Monday) {
     }
 
     @OptIn(DelicateIterableApi::class)
-    val nextRun : LocalDateTime?
+    val nextRun : T?
         get() = asIterable().firstOrNull()
 
     val expression: String
-        get() = partBuilders.values.joinToString(" ") { builder -> builder.value }
+        get() = partBuilders.values.joinToString(" ") { it.value }
 
     private val firstDayOfWeekIndex = WeekDays.entries.indexOf(firstDayOfWeek)
 
-    fun build(parts: Map<CronPart, PartValue>): Builder {
+    fun build(parts: Map<CronPart, PartValue>): Builder<T, D, P> {
         parts.forEach { entry -> partBuilders[entry.key]?.commonBuild(entry.value.type, entry.value.value) }
         return this
     }
 
-    fun seconds(type: TimeGroups, value: String) : Builder {
+    fun seconds(type: TimeGroups, value: String) : Builder<T, D, P> {
```

Comment:
When a function return this and not use lambda why don't use the scope function apply ?

```kotlin
fun seconds(type: TimeGroups, value: String) : Builder<T, D, P>  =  apply {
        partBuilders.getValue(CronPart.Seconds).commonBuild(type, value)
}
```

Reply:
To avoid recursive problems I kept `days` and `daysOfWeek` in old style.